### DISCUSSION TITLE
Distinct query not working due to pivot table information

### DISCUSSION THREAD

**Discussion by bradenroper** [adonisjs/core#1985] [Help]
Hello!

I have two tables, users and students, where users can be thought of as parents. They have a many-to-many relationship, and I want to (distinctly) get all of the grade levels that a parent may have students in.

Using `distinct()` in my query will not actually remove duplicate grade levels (ie. A parent with one student in 10th grade and two students in 9th grade will yield the results "9, 9, 10" instead of "9, 10"). After using `JSON.stringify()` on the results of the query, it seems to me that the issue is that the pivot table columns are included in the query even though they weren't explicitly selected, meaning the two "9th grade" records appear as different since they are results of different student records.

Assuming my issue is as I suspect, is there a way to exclude or remove the pivot table portion of the query or do I need to write out custom SQL or postprocess the object? I just feel like there may be a simpler solution that I am not seeing, thank you!

From my User model (the `gradeLevels()` function has the issue):
```
/**
* A parents have a many-to-many relationship with students
*/
students() {
	return this.belongsToMany('App/Models/Student').pivotTable('parent_student').pivotPrimaryKey(null)
}

/**
* Returns the grade levels for the students of this user
*/
gradeLevels() {
	return this.students().select('grade_level').distinct('grade_level')
}
```

The result of the query (after `JSON.stringify()`) is:
```
[
    {
        "grade_level": 9,
        "pivot":
        {
            "student_id": 1,
            "user_id": 2
        }
    },
    {
        "grade_level": 9,
        "pivot":
        {
            "student_id": 4,
            "user_id": 2
        }
    },
    {
        "grade_level": 10,
        "pivot":
        {
            "student_id": 2,
            "user_id": 2
        }
    }
]
```

**Comment by thetutlage** [comment]
Can you tell me what will be your custom SQL in this scenario and maybe I will be able to provide you the Lucid relationship equialavent of that

**Reply by bradenroper** [reply]
I had updated it to use this SQL. I suppose I could update this to use the Lucid functions for joins and such, I think my original issue is that I was just trying to use my `students()` relationship function, thinking I could just add the `distinct()` call to it.

```
  /**
   * Returns the grade levels for the students of this user
   */
  async gradeLevels() {
    const result = await Database.raw(`select distinct s.grade_level from users u
                            inner join parent_student ps on ps.user_id = u.id
                            inner join students s on ps.student_id = s.id
                            where u.id = ?
                            order by s.grade_level asc`, [this.id])
    return result.rows.map(r => r.grade_level)
  }
```
